Sources/Tracking/CustomerIO.swift

```diff
@@ -321,4 +340,42 @@ public class CustomerIO: CustomerIOInstance {
 
         identifyRepository.removeCustomer()
     }
+
+    /**
+     Track an event
+
+     [Learn more](https://customer.io/docs/events/) about events in Customer.io
+
+     - Parameters:
+     - name: Name of the event you want to track.
+     - data: Optional event body data
+     - onComplete: Asynchronous callback with `Result` of tracking an event.
+     Check result to see if error or success. Callback called on main thread.
+     - jsonEncoder: Provide custom JSONEncoder to have more control over the JSON request body
+     */
+    public func track<RequestBody: Encodable>(
+        name: String,
+        data: RequestBody,
+        timestamp: Date? = nil,
+        jsonEncoder: JSONEncoder? = nil,
+        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void
+    ) {
+        guard let identifyRepository = self.identifyRepository else {
+            return onComplete(Result.failure(.notInitialized))
+        }
+
+        identifyRepository
+            .trackEvent(name: name, data: data, timestamp: timestamp, jsonEncoder: jsonEncoder) { [weak self] result in
```

Comment:
What is the main use case for `timestamp` for a customer? Is it simply tracking the time that an event happened? 

If so, I propose that the default timestamp instead of `nil` (so the API records a time when the API request succeeded) is `Date()` to record when the event happened in the app.

Reply:
yeah I waffled a little on this one. In most of our SDKs it is treated as optional, because normally we don't really recommend using it. It also has the possibility of causing some [unexpected behaviour](https://github.com/customerio/services/blob/c630505b789742c6a0c05234f68acdb456605e9b/campaigns/match.go#L81) since we only allow events from the last 72h  to trigger a campaign.

That said, as a happy path & even with a bg queue, I think it's probably reasonable to default to `Date()` but still make it optional in case someone were to want it explicitly to be dated based on the receive time

Reply:
that said, I like _even more_ @clabland's idea that we follow the approach in our other SDKs and hide timestamp from the customer, but that we set the value if the request gets sent to the bg queue

Reply:
I see you [made a comment](https://github.com/customerio/customerio-ios/blob/6ffa57404c7cba73d6c4e4e01d129667bd0d0302/Sources/Tracking/CustomerIO.swift#L365) in the source code for doing this in the future. 

I also [made a note in the bg queue ticket](https://github.com/customerio/customerio-ios/issues/36) to do this. 

This is resolved!